### ISSUE TITLE
xonsh doesn't play well with dialog

### ISSUE THREAD

**Issue by FirstLoveLife** [xonsh/xonsh#3317]
## xonfig

<details>

```
$ xonfig
+------------------+-----------------+
| xonsh            | 0.9.8           |
| Python           | 3.7.4           |
| PLY              | 3.11            |
| have readline    | True            |
| prompt toolkit   | 2.0.9           |
| shell type       | prompt_toolkit2 |
| pygments         | 2.3.1           |
| on posix         | True            |
| on linux         | True            |
| distro           | unknown         |
| on darwin        | False           |
| on windows       | False           |
| on cygwin        | False           |
| on msys2         | False           |
| is superuser     | False           |
| default encoding | utf-8           |
| xonsh encoding   | utf-8           |
| encoding errors  | surrogateescape |
+------------------+-----------------+

```

</details>

## Expected Behavior

![image](https://user-images.githubusercontent.com/37442588/65410254-73841f80-de1c-11e9-9649-958fecae96eb.png)

curse can be moved up/down
## Current Behavior

![image](https://user-images.githubusercontent.com/37442588/65410281-84cd2c00-de1c-11e9-8c41-5c36ea81b3c5.png)

Up/Down will be mapped into ^[OA/^[OB
### Traceback (if applicable)

<details>

```
traceback
```

</details>

## Steps to Reproduce

download linux kernel source
make defconfig
make menuconfig

**Comment by laloch** [comment]
Hi @FirstLoveLife, thanks for reporting.
You can work the issue around by setting `make` to run unthreaded temporarily:
```
__xonsh__.commands_cache.threadable_predictors['make'] = lambda args: False`
```

**Comment by laloch** [comment]
Another possible work around would be to run `make menuconfig` uncaptured:
```
$ $[make menuconfig]
```
or to create an uncaptured alias for it:
```
$ aliases["linux-menuconfig"] = "$[make menuconfig]"
$ linux-menuconfig
```

**Comment by FirstLoveLife** [comment]
@laloch Thanks for your workaround! But i want to know why does xonsh have such problems?

**Comment by cryzed** [comment]
Having issues with this too. I also don't know why xonsh has problems with this, but other shells don't need special workaround code -- I'm not trying to be rude, I just admit I never looked into it.

**Comment by scopatz** [comment]
It is because the interactive program is being hidden in make

**Comment by cryzed** [comment]
Well I was more confused about the mechanics behind it: why does xonsh need to be aware of what is considered an interactive program? Do other shells also just have a list of programs that have to be handled in a specific way for them not to break?

**Comment by scopatz** [comment]
No, no other shells have this.  This is a by product of a couple of technical issues. 

1. We want to be able to capture stdout/stderr when approriate (ie no other shell has this)
2. We want to be fully interactive

Because of those we try to run on (python) threads.  This can cause issues for programs where no stdout capturing is really wanted or needed.  We try to figure this out ahead of time (because you can't switch once the program has started), but our heuristics fail in a couple of cases. Like most people run make in a way that prints output to the terminal.  Sometimes make is run in a way that uses ncurses. I think in this case we could special case to see if `menuconfig` is one of the command line arguments

**Comment by cryzed** [comment]
Hey thanks for the detailed explanation!

**Comment by g33kex** [comment]
Hi,
There is the same broken behavior with other applications using ncurses, such as `alsamixer`. But using the workaround `aliases["alsamixer"] = "$[/usr/bin/alsamixer]"` works.

**Comment by anki-code** [comment]
Closed with reference to #4214
